src/Microsoft.DotNet.Interactive.Kql/KqlKernelExtension.cs

```diff
@@ -3,56 +3,41 @@
 
 using System;
 using System.IO;
+using System.Linq;
 using System.Runtime.InteropServices;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Html;
 using Microsoft.DotNet.Interactive.SqlServer;
+using Microsoft.DotNet.Interactive.Utility;
 
 namespace Microsoft.DotNet.Interactive.Kql
 {
     public class KqlKernelExtension : IKernelExtension
     {
-        public Task OnLoadAsync(Kernel kernel)
+        public async Task OnLoadAsync(Kernel kernel)
         {
             if (kernel is CompositeKernel compositeKernel)
             {
-                var root = compositeKernel.RootKernel.FindResolvedNativeSqlToolsServicePackageReference();
-                if (root != null)
+                // Check if the required Sql Tools Service tool is installed, and then install it if necessary
+                var dotnet = new Dotnet();
+                var installedGlobalTools = await dotnet.ToolList();
+                const string kqlToolName = "MicrosoftKustoServiceLayer";
+                bool kqlToolInstalled = installedGlobalTools.Any(tool => string.Equals(tool, kqlToolName, StringComparison.InvariantCultureIgnoreCase));
+                if (!kqlToolInstalled)
                 {
-                    var pathToService = root.PathToToolsService("MicrosoftKustoServiceLayer");
+                    var commandLineResult = await dotnet.ToolInstall("Microsoft.SqlServer.KustoServiceLayer.Tool", null, null, "1.0.0");
```

Comment:
It seems like we shouldn't hard code the version. How often do we expect this to be updated?

Reply:
It doesn't have a dedicated release cadence, so any time we have any useful improvements, basically. Diego wanted a fixed version here in case of breaking changes in the future.